File: 53. Maximum Subarray/53. Maximum Subarray.rb
# @param {Integer[]} nums
# @return {Integer}
def max_sub_array(nums)
    sum, max = nums[0], nums[0]
    
    for val in nums[1..-1] do
        sum = val + (sum > 0 ? sum : 0)
        max = [sum,max].max
    end
    max
end

File: 53. Maximum Subarray/53. Maximum Subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    var sum = nums[0];
    var max = sum;
    
    for(var i = 1; i<nums.length; i++) {
        sum = nums[i] + (sum > 0 ? sum : 0);
        max = Math.max(sum, max);
    }
    return max;
};
